distutils/dist.py

```diff
@@ -1189,7 +1191,10 @@ def get_version(self):
         return self.version or "0.0.0"
 
     def get_fullname(self):
-        return f"{self.get_name()}-{self.get_version()}"
+        return "{}-{}".format(
+            canonicalize_name(self.get_name()).replace('-', '_'),
+            canonicalize_version(self.get_version()),
```

Comment:
```suggestion
            self.get_version(),
```
To avoid problems with trailing zeros as in https://github.com/pypa/setuptools/commit/df45427cbb67c1149fcf5d2d1e2705e69b3baf0c